python/ray/data/_internal/block_batching/util.py

```diff
@@ -155,12 +161,22 @@ def format_batches(
     Returns:
         An iterator over batch index and the formatted batch.
     """
+    last_stats_update_time = 0
```

Comment:
I think this needs to be a thread-local variable.

Reply:
Why should each thread have their own timer? Aren't they reporting global stats?

Reply:
Ah my mistake, yes you're right it should be a global variable. But we do need to make it thread-safe.

Reply:
This here was actually already thread-local, I've updated it to be non-local